src/collector.py

```diff
@@ -0,0 +1,451 @@
+                ("status", _get_overall_status_metric(api_metrics)),
+                ("current_connections", _get_current_connections_metric(api_metrics)),
+                ("up_time", _get_up_time_metric(api_metrics)),
+                ("event_loop_delay", _get_event_loop_delay_metric(api_metrics)),
+                ("heap_total", _get_heap(api_metrics, Heap.TOTAL)),
+                ("heap_used", _get_heap(api_metrics, Heap.USED)),
+                ("heap_size", _get_heap(api_metrics, Heap.SIZE)),
+                ("re_set_size", _get_resident_set_size(api_metrics)),
+                ("load_1m", _get_load(api_metrics, Load.ONE_M)),
+                ("load_5m", _get_load(api_metrics, Load.FIVE_M)),
+                ("load_15m", _get_load(api_metrics, Load.FIFTEEN_M)),
+                ("os_mem_total", _get_os_mem(api_metrics, Memory.TOTAL)),
+                ("os_mem_free", _get_os_mem(api_metrics, Memory.FREE)),
+                ("os_mem_used", _get_os_mem(api_metrics, Memory.USED)),
+                ("resp_time_avg", _get_resp_time(api_metrics, Response.AVG)),
+                ("resp_time_max", _get_resp_time(api_metrics, Response.MAX)),
+                ("req_disconnects", _get_req(api_metrics, RequestsCount.DISCONNECTS)),
+                ("req_total", _get_req(api_metrics, RequestsCount.TOTAL)),
+            ] + [("statuses", status) for status in _get_statuses_metrics(api_metrics)]
+        return []
+
+
+def collect_api_status(config: Config) -> dict:
+    """Use the dashboard API to get the status of the dashboard.
+
+    Args:
+        config (Config): Config necessary to access the information.
+
+    Returns:
+        dict: Raw response from the API
+    """
+    auth = HTTPBasicAuth(config.user, config.password)
+    headers = {"Accept": "application/json"}
+    url = f"{config.url}{API_STATUS_ENDPOINT}"
+
+    try:
+        with requests.Session() as session:
+            response = session.get(
+                url,
+                auth=auth,
+                verify=False,
+                headers=headers,
+                timeout=5,
+            )
+        response.raise_for_status()
+        return json.loads(response.text)
+    except HTTPError as e:
+        logger.error(
+            "Request to %s status code: %s response text %s.",
+            url,
+            e.response.status_code,
+            e.response.text,
+        )
+    except Timeout:
+        logger.error("Request to %s timed out.", url)
+    except RequestException as e:
+        logger.error("It was not possible to collect the OpenSearch Dashboards metrics %s", e)
+
+    return {}
+
+
+def _get_overall_status_metric(api_metrics: dict) -> Optional[Metric]:
+    """Get the overall status of the dashboards.
+
+    Args:
+        api_metrics (dict): Response from the API
+
+    Returns:
+        Optional[Metric]: Prometheus Gauge metric if the metric exist in the API
+    """
+    metric_name = f"{METRICS_PREFIX}status"
+    match api_metrics:
```

Comment:
Note: Use of the match statement locks the code into Python 3.10 or newer, i.e. compatible with Jammy 24.04's Python or newer, but not compatible with the Python 3.8 of 22.04 Focal.  Although, based on earlier code review discussion, this seems to be known and intended.